diff --git a/src/Store.js b/src/Store.js
index a96b8ea..0f94398 100644
--- a/src/Store.js
+++ b/src/Store.js
@@ -220,7 +220,7 @@ class Store {
     this.events.emit('ready', this.address.toString(), this._oplog.heads)
   }
 
-  sync (heads) {
+  async sync (heads) {
     this._stats.syncRequestsReceieved += 1
     logger.debug(`Sync request #${this._stats.syncRequestsReceieved} ${heads.length}`)
     if (heads.length === 0) {
@@ -261,10 +261,19 @@ class Store {
       return head
     }
 
-    return mapSeries(heads, saveToIpfs)
-      .then(async (saved) => {
-        return this._replicator.load(saved.filter(e => e !== null))
+    const saved = await mapSeries(heads, saveToIpfs)
+    await this._replicator.load(saved.filter(e => e !== null))
+
+    return new Promise(resolve => {
+      this.events.on('replicate.progress', (address, hash, entry, progress, have) => {
+        if (progress === have) {
+          this.events.on('replicated', resolve)
+        }
       })
+      if (!this._replicator._buffer.length && !Object.values(this._replicator._queue).length) {
+        resolve()
+      }
+    })
   }
 
   loadMoreFrom (amount, entries) {
