diff --git a/src/modules/core/sagas/setupUserContext.ts b/src/modules/core/sagas/setupUserContext.ts
index 84b0f8fc7..44ad97729 100644
--- a/src/modules/core/sagas/setupUserContext.ts
+++ b/src/modules/core/sagas/setupUserContext.ts
@@ -1,5 +1,5 @@
 import { all, call, fork, put } from 'redux-saga/effects';
-import { formatEther } from 'ethers/utils';
+import { formatEther, bigNumberify } from 'ethers/utils';

 import { WalletMethod } from '~immutable/index';
 import { createAddress } from '~utils/web3';
@@ -221,6 +221,9 @@ export default function* setupUserContext(
      */
     yield updateNetworkContracts();

+    console.log(colonyManager);
+    console.log(bigNumberify);
+
     setupOnBeforeUnload();
