diff --git src/history.js src/history.js
index 636d40f..021f1bf 100644
--- src/history.js
+++ src/history.js
@@ -2,6 +2,8 @@
 
 import { createBrowserHistory } from 'history';
 
-const history = createBrowserHistory();
+const history = createBrowserHistory({
+  basename: '/dapp',
+});
 
 export default history;
diff --git webpack.config.js webpack.config.js
index f7c2a6f..550d62b 100644
--- webpack.config.js
+++ webpack.config.js
@@ -32,7 +32,7 @@ const config = {
   output: {
     filename: 'bundle.js',
     path: path.resolve(__dirname, 'dist'),
-    publicPath: '/',
+    publicPath: '/dapp',
   },
   resolve: {
     alias: Object.assign(
